Source/Core/DolphinQt2/Config/Mapping/MappingButton.cpp

```diff
@@ -2,24 +2,36 @@
 // Licensed under GPLv2+
 // Refer to the license.txt file included.
 
+#include <thread>
+
 #include <QMouseEvent>
 #include <QRegExp>
 #include <QString>
-#include <thread>
 
 #include "DolphinQt2/Config/Mapping/MappingButton.h"
 
 #include "Common/Thread.h"
+#include "DolphinQt2/Config/Mapping/IOWindow.h"
+#include "DolphinQt2/Config/Mapping/MappingCommon.h"
 #include "DolphinQt2/Config/Mapping/MappingWidget.h"
 #include "DolphinQt2/Config/Mapping/MappingWindow.h"
 #include "InputCommon/ControlReference/ControlReference.h"
 #include "InputCommon/ControllerEmu/ControllerEmu.h"
 #include "InputCommon/ControllerInterface/ControllerInterface.h"
+#include "InputCommon/ControllerInterface/Device.h"
+
+static QString TruncateDisplayString(const QString& string)
```

Comment:
Here's an example of the issue with eliding based on number of characters rather than on available width:

![image](https://cloud.githubusercontent.com/assets/594093/26607138/e51cfd10-4549-11e7-955a-92506f126194.png)

If you want, I can go figure out the ElidedButton stuff and put up a separate PR, since this is already a problem with current controller mapping window (in that it doesn't elide at all, and my mappings are giant-ass strings like `Up Arrow + Input/0/Xbox One Wired Controller:Axis Y-`).